packages/devtools_shared/lib/src/memory/adb_memory_info.dart

```diff
@@ -17,17 +17,41 @@ class AdbMemoryInfo {
     this.total,
   );
 
-  factory AdbMemoryInfo.fromJson(Map<String, dynamic> json) => AdbMemoryInfo(
-        json[realTimeKey] as int,
-        json[javaHeapKey] as int,
-        json[nativeHeapKey] as int,
-        json[codeKey] as int,
-        json[stackKey] as int,
-        json[graphicsKey] as int,
-        json[otherKey] as int,
-        json[systemKey] as int,
-        json[totalKey] as int,
-      );
+  /// @param rawData (from service extension) if true then all data returned from
```

Comment:
In Dart rather than using `@param rawData` write the comments, use `[rawData]`

Instead of having to pass in `rawData`, consider instead turning these all into private fields with public getters that multiply by 1024. That way you wouldn't have to worry about any possible bugs passing in the wrong value for rawData. if you leave rawData, consider changing the name to make it clearer what the effect is. For example `inKilobytes` or similar.

Reply:
I think this might be better.  Added two constructors for fromJsonInKB and fromJson.

```
  /// All data inside of AdbMemoryInfo is in total bytes. When receiving ADB data
  /// from the service extension (directly from ADB) then the data is in kilobytes.
  /// See the factory constructor fromJsonInKB.
  factory AdbMemoryInfo.fromJson(Map<String, dynamic> json) 

  /// Use when converting data received from the service extension, directly from
  /// ADB. All data received from ADB dumpsys meminfo is in kilobytes must adjust to
  /// total bytes for AdbMemoryInfo data.
  factory AdbMemoryInfo.fromJsonInKB(
    Map<String, dynamic> json,
  ) 

```